### PR TITLE
[FLINK-33805] Implement restore tests for OverAggregate node

### PR DESCRIPTION
**Pull request by jnh5y**
## What is the purpose of the change

Implement restore tests for OverAggregate node.

## Verifying this change

This change added tests and can be verified as follows:

* Added restore tests for OverAggregate node which verifies the generated compiled plan with the saved compiled plan

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import org.apache.flink.table.test.program.SourceTestStep;
+import org.apache.flink.table.test.program.TableTestProgram;
+import org.apache.flink.types.Row;
+
+import static org.apache.flink.table.api.config.TableConfigOptions.LOCAL_TIME_ZONE;
+
+/** {@link TableTestProgram} definitions for testing {@link StreamExecOverAggregate}. */
+public class OverAggregateTestPrograms {
+
+    static final Row[] DATA = {
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java:32]: Can we please fix the visibility modifiers in the entire class?

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+                    BEFORE_RESTORE_DATA,
+                    AFTER_RESTORE_OUTPUT);
+
+    static final TableTestProgram OVER_AGGREGATE_BOUNDED_NON_PARTITIONED_ROWS =
+            getTableTestProgram(
+                    "over-aggregate-bounded-non-partitioned-rows",
+                    "validates over aggregate node",
+                    "PARTITION BY c ORDER BY rowtime RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW",
+                    BEFORE_RESTORE_DATA,
+                    AFTER_RESTORE_OUTPUT);
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java:130]: Are those two different? I can't see a difference in queries. Could you also update the descriptions, they're also the same now.

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        "+I[Hello World, 140, 1970-01-01T00:02:20, 18, 3, 3, 21]",
+        "+I[Hello World, 150, 1970-01-01T00:02:30, 8, 4, 4, 29]",
+        "+I[Hello World, 200, 1970-01-01T00:03:20, 20, 5, 5, 49]"
+    };
+
+    public static final String[] AFTER_RESTORE_DATA_PRECEDING_ROWS = {
+        "+I[Hello World, 202, 1970-01-01T00:03:22, 20, 6, 6, 69]"
+    };
+
+    static final TableTestProgram OVER_AGGREGATE_BOUNDED_PARTITIONED_PRECEDING_ROWS =
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java:189]: Is `Preceding` differentiating enough? I believe the difference is `time-based` preceding vs `count-based` preceding.

**Review by dawidwys — CHANGES_REQUESTED** [review]

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java (hunk 1 above)
Comment (jnh5y) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/OverAggregateTestPrograms.java:32]: Yes.

**Review by dawidwys — APPROVED** [review]
LGTM! Thank you for the update!

**Pull request closed without merging**